### ISSUE TITLE
Feature Request: Variable boundaries like SUPPA

### ISSUE THREAD

**Issue by Alex-Nesta** [timbitz/Whippet.jl#99]
I've seen a few issues posted here that describe lots of AF and AL events. These could simply be due to a noisy reference gtf. SUPPA makes it easy to limit this issue by defining a variable boundary region for first and last exons. Can this feature be added to whippet?

**Comment by JamalEH** [comment]
Hi Alex,
I experienced exactly the same thing in my datasets (polyA+ unstranded RNA-seq datasets), but I never get feedbacks on these subset of abundant and highly significant events. I agree with you what you are proposing here could be an option to get red of these events from Whippet's output files.

**Comment by timbitz** [comment]
@Alex-Nesta Yeah these are due to the reference GTF -- Sure this could be added, but I doubt I'll get to it in this version of Whippet.  I think your best bet is to just filter out the events that you think should be biologically meaningful or not.  

@JamalEH One possibility for the reason you're getting lots of significant TS/TE events but not splicing events is that you have a low read-depth dataset, and since splicing events for whippet require exon-exon junction spanning reads by default, whereas TE/TS events utilize exon-body reads, the depth required for statistical significance of these two types is fundamentally different.  That doesn't mean that TS/TE events are more biologically prevalent in your sample, but perhaps more technically prevalent given your sequencing depth.

**Comment by JamalEH** [comment]
@timbitz 
Thank you so much for your feedback!

Actually, I observed a similar trend of TS/TE events frequency in a dataset of 150M reads per library, on average. The dataset is paired-end, stranded, with a read length of 76bp. I observed a prevalence of TS/TE events for this dataset as well, and the problem is that, it was very difficult to see those events from IGV genome viewer (with the exception of some easley seen events). I should however, say that for the previous dataset, the TE events I observed almost 55% of which overlapped with the CLIP-seq coordinates showing the binding of a splicing factor that was investigated in my study. This splicing factor is well known to regulate the usage of alternative polyadenylation sites.

I have a related question, why do we observe a very low overlap with other tools, such as SUPPA2, rMATS, PSIsigma, when considering the significance of the dPSI. What could be in your opinion the best way to overlap the results of 2 different tools? This is true for all the splicing event types, including alternative first and last exons. The trend of splicing change (e.g. dPSI) among conditions is similar between the tools, but when filtering those events based on the significance, the overlap (events significant in 2 tools) is extremely low, even if the same gene annotation model and the same read files are analysed!

Thank you and my apologies for the late feedback!

Best regards,
Jamal.

**Comment by santataRU** [comment]
> @timbitz Thank you so much for your feedback!
> 
> Actually, I observed a similar trend of TS/TE events frequency in a dataset of 150M reads per library, on average. The dataset is paired-end, stranded, with a read length of 76bp. I observed a prevalence of TS/TE events for this dataset as well, and the problem is that, it was very difficult to see those events from IGV genome viewer (with the exception of some easley seen events). I should however, say that for the previous dataset, the TE events I observed almost 55% of which overlapped with the CLIP-seq coordinates showing the binding of a splicing factor that was investigated in my study. This splicing factor is well known to regulate the usage of alternative polyadenylation sites.
> 
> I have a related question, why do we observe a very low overlap with other tools, such as SUPPA2, rMATS, PSIsigma, when considering the significance of the dPSI. What could be in your opinion the best way to overlap the results of 2 different tools? This is true for all the splicing event types, including alternative first and last exons. The trend of splicing change (e.g. dPSI) among conditions is similar between the tools, but when filtering those events based on the significance, the overlap (events significant in 2 tools) is extremely low, even if the same gene annotation model and the same read files are analysed!
> 
> Thank you and my apologies for the late feedback!
> 
> Best regards, Jamal.

I just came across one paper about a pipeline that "performs alternative splicing analysis using rMATS and Whippet with subsequent overlapping of the results, enabling robust splicing analysis with only one command". 
https://github.com/didrikolofsson/rmappet/

https://www.sciencedirect.com/science/article/pii/S0006291X23002280?via%3Dihub

**Comment by santataRU** [comment]
Hi, Jamal,

I have also observed a prevalence of TS/TE events, and I checked a few of them (d|PSI|>0.1 and Probability > 0.95) in IGV, and I did not see an obvious pattern in IGV supporting these events.

Xiao

> I observed a prevalence of TS/TE events for this dataset as well, and the problem is that, it was very difficult to see those events from IGV genome viewer (with the exception of some easley seen events).
